packages/pyright-internal/src/pyright.ts

```diff
@@ -472,33 +475,76 @@ async function runSingleThreaded(
             return;
         }
 
+        const rootDir =
+            typeof options.executionRoot === 'string' || options.executionRoot === undefined
+                ? Uri.file(options.executionRoot ?? '', service.serviceProvider)
+                : options.executionRoot;
+        const allDiagnostics = results.diagnostics;
+        const baselinedErrorCount = Object.values(getBaselinedErrors(rootDir).files).flatMap((file) => file).length;
+        const newErrorCount = allDiagnostics
+            .map(
+                (file) =>
+                    file.diagnostics.filter(
+                        (diagnostic) =>
+                            !isHintDiagnostic(diagnostic) || diagnostic.baselineStatus === 'baselined with hint'
+                    ).length
+            )
+            .reduce(add);
+        const filteredDiagnostics = results.diagnostics.map((file) => ({
+            ...file,
+            diagnostics: file.diagnostics.filter((diagnostic) => !diagnostic.baselineStatus),
+        }));
+
+        // if there are any unbaselined errors, don't write to the baseline unless the user explicitly passed
+        // --writebaseline
+        if (
+            args.writebaseline ||
+            !filteredDiagnostics.map((fileWithDiagnostics) => fileWithDiagnostics.diagnostics.length).reduce(add)
```

Comment:
don't write baseline if it contains files that weren't included when it ran, to prevent users from accidentally deleting files from the baseline.

alternatively it could just do what the lsp does and only write files it can see, and leave everything else as is

Reply:
the problem with that approach is if a file gets deleted/renamed its baselined errors will just stick around